.github/workflows/quicktest.yml

```diff
@@ -0,0 +1,88 @@
+  workflow_dispatch:
+
+jobs:
+  #### QUICK TEST STAGE ####
+  # This is a much quicker test which only runs the unit tests and linting against low/high
+  # supported PHP/PHPCS/WPCS combinations.
+  quicktest:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        include:
+          - php: 'latest'
+            phpcs_version: 'dev-master'
+            wpcs_version: 'dev-master'
+          - php: 'latest'
+            phpcs_version: '3.6.0'
+            wpcs_version: '2.3.0'
+          - php: '5.4'
+            phpcs_version: 'dev-master'
+            wpcs_version: '2.3.0'
+          - php: '5.4'
+            phpcs_version: '3.6.0'
+            wpcs_version: 'dev-master'
+
+    name: "QTest${{ matrix.phpcs_version == 'dev-master' && ' + Lint' || '' }}: PHP ${{ matrix.php }} - PHPCS ${{ matrix.phpcs_version }} - WPCS ${{ matrix.wpcs_version }}"
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      # On stable PHPCS versions, allow for PHP deprecation notices.
+      # Unit tests don't need to fail on those for stable releases where those issues won't get fixed anymore.
+      - name: Setup ini config
+        id: set_ini
+        run: |
+          if [ "${{ matrix.phpcs_version }}" != "dev-master" ]; then
+            echo '::set-output name=PHP_INI::error_reporting=E_ALL & ~E_DEPRECATED, display_errors=On'
+          else
+            echo '::set-output name=PHP_INI::error_reporting=E_ALL, display_errors=On'
+          fi
+
+      - name: Install PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          ini-values: ${{ steps.set_ini.outputs.PHP_INI }}
+          coverage: none
+
+      - name: 'Composer: adjust dependencies'
+        run: |
+          # Set the PHPCS version to test against.
+          composer require --no-update --no-scripts squizlabs/php_codesniffer:"${{ matrix.phpcs_version }}"
+          # Set the WPCS version to test against.
+          composer require --no-update --no-scripts wp-coding-standards/wpcs:"${{ matrix.wpcs_version }}"
+
+      # Install dependencies and handle caching in one go.
+      # @link https://github.com/marketplace/actions/install-composer-dependencies
+      - name: Install Composer dependencies - normal
+        if: ${{ startsWith( matrix.php, '8' ) == false && matrix.php != 'latest' }}
+        uses: ramsey/composer-install@v1
+
+      # For the PHP 8.0 and higher, we need to install with ignore platform reqs as not all dependencies allow it.
+      - name: Install Composer dependencies - with ignore platform
+        if: ${{ startsWith( matrix.php, '8' ) || matrix.php == 'latest' }}
+        uses: ramsey/composer-install@v1
+        with:
+          composer-options: --ignore-platform-reqs
+
+      - name: Verify installed standards
+        run: vendor/bin/phpcs -i
+
+      - name: Lint against parse errors
+        if: matrix.phpcs_version == 'dev-master'
```

Comment:
Have you considered adding a separate value to trigger linting?
This is a very implicit consequence. Perhaps linting isn't always needed in the future for dev-master?

Also, adding an additional property to the matrix doesn't seem like a big deal.

Reply:
See my answer here: https://github.com/Yoast/yoastcs/pull/250#discussion_r660789383